sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -468,73 +535,38 @@ export class StreamingReceiver extends MessageReceiver {
 
       const translatedError = receiverError ? translate(receiverError) : receiverError;
 
-      // Track-1
-      //   - We should only attempt to reopen if either no error was present,
-      //     or the error is considered retryable.
-      // Track-2
-      //  Reopen
-      //   - If no error was present
-      //   - If the error is a MessagingError and is considered retryable
-      //   - Any non MessagingError because such errors do not get
-      //     translated by `@azure/core-amqp` to a MessagingError
-      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
-      const shouldReopen =
-        translatedError instanceof MessagingError ? translatedError.retryable : true;
-
-      // Non-retryable errors that aren't caused by disconnect
-      // will have already been forwarded to the user's error handler.
-      // Swallow the error and return quickly.
-      if (!shouldReopen && !causedByDisconnect) {
-        logger.logError(
-          translatedError,
-          "%s Encountered a non retryable error on the receiver. Cannot recover receiver. encountered error",
-          this.logPrefix,
-          this.name,
-          this.address
-        );
-        return;
-      }
+      logger.logError(
+        translatedError,
+        `${this.logPrefix} Encountered an error on the connection. Reinitializing receiver.`
+      );
 
-      // Non-retryable errors that are caused by disconnect
-      // haven't had a chance to show up in the user's error handler.
-      // Rethrow the error so the surrounding try/catch forwards it appropriately.
-      if (!shouldReopen && causedByDisconnect) {
-        logger.logError(
-          translatedError,
-          "%s Encountered a non retryable error on the connection. Cannot recover receiver.",
-          this.logPrefix,
-          this.name,
-          this.address
-        );
-        throw translatedError;
+      if (translatedError && this._onError) {
+        this._onError({
```

Comment:
When there is an error on the AMQP link or session, we call the user's error handler. This is soon followed by the AMQP link or session closing, which is when we call the `onDetached()` method. With this change, we end up calling the user's error handler twice for the same error.

Reply:
I'll take a look at this - I've got some other ideas around where to call onError based on some of your other feedback.

Reply:
Does it mean that calling onError only in onDetached is enough?

Reply:
I made a change that I think is the "right" thing to do here but let's see. I've made it now so that the error that is reported to onDetached is logged (in onDetached itself) _but_ we don't report it to the user's onError handler.

My reasoning is that the user isn't going to be able to do anything with _that_ error (or at least, it represents an error condition that might not matter) _but_ the next error (the one potentially coming from init()) will be useful to them and that is reported to their onError() handler.

Does this address yours (@ramya-rao-a ) and @HarshaNalluru 's concerns?

Reply:
Makes sense, since onDetached would anyway call init, and since we call onError from there.

Reply:
That sounds good, but then you will need to remove the call to the user's error handler from the AMQP link & session error event handler